source/geometries/JaszczakPhantom.cc

```diff
@@ -0,0 +1,200 @@
+  auto radius_pos = cylinder_inner_diam_/4.;
+  auto z_pos      = - cylinder_height_/2. + sphere_height_;
+
+  for (unsigned long i=0; i<sphere_radii.size(); i++) {
+    BuildSpheres(i, sphere_radii[i], radius_pos, z_pos, water_logic, water);
+  }
+
+  // Rods
+  std::vector<G4double> rod_radii =
+    {rod1_d_/2, rod2_d_/2, rod3_d_/2, rod4_d_/2, rod5_d_/2, rod6_d_/2};
+  z_pos = - cylinder_height_/2. + rod_height_/2;
+  for (unsigned long i=0; i<rod_radii.size(); i++) {
+    BuildRods(i, rod_radii[i], z_pos, water_logic, water);
+  }
+
+  // Relative actvities
+  auto max_activity = std::max(sphere_activity_, std::max(bckg_activity_, rod_activity_));
+  bckg_activity_   /= max_activity;
+  sphere_activity_ /= max_activity;
+  rod_activity_    /= max_activity;
+
+  G4cout << "*** Relative activities (background, spheres, rods) ***" << G4endl;
+  G4cout << bckg_activity_ << ", " << sphere_activity_  << ", " << rod_activity_ << G4endl;
+
+}
+
+
+void JaszczakPhantom::BuildSpheres(unsigned long n, G4double r, G4double r_pos, G4double z_pos,
+                                   G4LogicalVolume* mother_logic, G4Material* mat) const
+{
+  auto sphere_name = "SPHERE" + std::to_string(n);
+  auto sphere_solid = new G4Orb(sphere_name, r);
+  auto sphere_logic = new G4LogicalVolume(sphere_solid, mat, sphere_name);
+  auto angle = n * 60 * deg;
+  auto x_pos = r_pos * cos(angle);
+  auto y_pos = r_pos * sin(angle);
+  new G4PVPlacement(0, G4ThreeVector(x_pos, y_pos, z_pos), sphere_logic,
+                    sphere_name, mother_logic, false, 0, true);
+}
+
+
+
+void JaszczakPhantom::BuildRods(unsigned long n, G4double r, G4double z_pos,
+                                G4LogicalVolume* mother_logic, G4Material* mat) const
+  {
+    auto diam = 2 * r;
+
+    // Sector displacement from centre, to accommodate gap between sectors
+    auto gap =  14.4 * mm;
+    auto dx  = gap * cos(pi/6);
+    auto dy  = gap * sin(pi/6);
+    // Displacement of first rod WRT sector corner
+    dx += r * sqrt(3);
+    dy += r;
+    // Basis vectors of rod lattice
+    const auto Ax = 2.0, Ay = 0.0;
+    const auto Bx = 1.0, By = sqrt(3);
+    auto a = 0;
+    for (bool did_b=true ; did_b; a+=1) {
+      did_b = false;
+      for (auto b = 0; /*break when touches the cylinder*/; b+=1, did_b = true) {
+        auto x_pos = (a*Ax + b*Bx) * diam + dx;
+        auto y_pos = (a*Ay + b*By) * diam + dy;
+        auto margin = 0.1 * mm;
+        if (sqrt(x_pos*x_pos + y_pos*y_pos) + r + margin >= cylinder_inner_diam_/2.) { break; }
+        auto label = "ROD" + std::to_string(n);
+        auto rod_solid = new G4Tubs(label, 0, r, rod_height_/2, 0, twopi);
+        auto rod_logic = new G4LogicalVolume(rod_solid, mat, label);
+        G4ThreeVector pos = G4ThreeVector(x_pos, y_pos, z_pos);
+        G4ThreeVector newpos = pos.rotateZ(n*pi/3);
+        new G4PVPlacement(0, newpos, rod_logic, label, mother_logic, false, 0, true);
+
```

Comment:
<img width="327" alt="Screenshot 2022-07-07 at 11 43 34" src="https://user-images.githubusercontent.com/32193826/177744344-1be407be-c0b5-42cb-99a1-63ae1deb0300.png">

Is the rod alone in the top right part on purpose?

Reply:
I have looked this kind of phantoms in different companies' webpages and it seems that none of them have the configuration of the triangle with the smallest diameters. See, for instance, https://floridainstrumentacion.es/floridawp/wp-content/uploads/2017/12/Flanged_Jaszczak_Phantoms.pdf, ultra-deluxe model. It seems that the last row of rods is left out, right? I've checked the inner diameter of the cylinder and it's correct. I'll have a closer look and try to implement the geometry of this model.

Reply:
Have a look now, and tell me what you think!

Reply:
> I have looked at this kind of phantoms in different companies' webpages and it seems that none of them have the configuration of the triangle with the smallest diameters.

Yes, that's what I saw yesterday and because of that I asked about the model.

>  It seems that the last row of rods is left out, right?

Yes...

Reply:
> Have a look now, and tell me what you think!

Perfect!